From d485a1d1bd298d1eef6b65aa693ebb0994316f9c Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Wed, 5 Jan 2011 10:13:08 +1000
Subject: [PATCH 12/18] Change kernel version check to simple ifdefs for MT defines.

The first few were introduced in the same commit (5e5ee68), the others
separately. Use a simple ifdef instead of a kernel version check because
some distros may have backported bits and pieces without changing the
version number.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 evtest.c |    8 +++++---
 1 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/evtest.c b/evtest.c
index 06f7948..c0d1f15 100644
--- a/evtest.c
+++ b/evtest.c
@@ -309,7 +309,7 @@ char *absolutes[ABS_MAX + 1] = {
 	[ABS_DISTANCE] = "Distance",	[ABS_TILT_X] = "XTilt",
 	[ABS_TILT_Y] = "YTilt",		[ABS_TOOL_WIDTH] = "Tool Width",
 	[ABS_VOLUME] = "Volume",	[ABS_MISC] = "Misc",
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,30)
+#ifdef ABS_MT_BLOB_ID
 	[ABS_MT_TOUCH_MAJOR] = "Touch Major",
 	[ABS_MT_TOUCH_MINOR] = "Touch Minor",
 	[ABS_MT_WIDTH_MAJOR] = "Width Major",
@@ -319,12 +319,14 @@ char *absolutes[ABS_MAX + 1] = {
 	[ABS_MT_POSITION_Y] = "Position Y",
 	[ABS_MT_TOOL_TYPE] = "Tool Type",
 	[ABS_MT_BLOB_ID] = "Blob ID",
+#endif
+#ifdef ABS_MT_TRACKING_ID
 	[ABS_MT_TRACKING_ID] = "Tracking ID",
 #endif
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,35)
+#ifdef ABS_MT_PRESSURE
 	[ABS_MT_PRESSURE] = "Pressure",
 #endif
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,36)
+#ifdef ABS_MT_SLOT
 	[ABS_MT_SLOT] = "Slot",
 #endif
 
-- 
1.7.3.1

